presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcResultSetTimezone.java

```diff
@@ -0,0 +1,384 @@
+    {
+        try (ResultSet rs = prestoQuery(expression, sessionTimezoneId); ResultSet referenceResultSet = reference.query(expression, sessionTimezoneId)) {
+            assertTrue(rs.next());
+            assertTrue(referenceResultSet.next());
+            assertion.accept(rs, referenceResultSet, 1);
+            assertFalse(rs.next());
+            assertFalse(referenceResultSet.next());
+        }
+    }
+
+    private ResultSet prestoQuery(String expression, String timezoneId)
+            throws Exception
+    {
+        connection.unwrap(PrestoConnection.class).setTimeZoneId(timezoneId);
+        return statement.executeQuery("SELECT " + expression);
+    }
+
+    private Calendar getCalendar()
+    {
+        return Calendar.getInstance(TimeZone.getTimeZone(OTHER_TIMEZONE));
+    }
+
+    private ZoneId getZoneId()
+    {
+        return ZoneId.of(getCalendar().getTimeZone().getID());
+    }
+
+    private interface ReferenceDriver
+    {
+        ResultSet query(String expression, String timezoneId) throws Exception;
+
+        void setup();
+
+        void teardown();
+
+        void close();
+    }
+
+    private static class OracleReferenceDriver
+            implements ReferenceDriver
+    {
+        private OracleContainer oracleServer;
+        private Connection connection;
+        private Statement statement;
+
+        OracleReferenceDriver()
+        {
+            oracleServer = new OracleContainer("wnameless/oracle-xe-11g-r2");
+            oracleServer.start();
+        }
+
+        @Override
+        public ResultSet query(String expression, String timezoneId)
+                throws Exception
+        {
+            statement.execute(format("ALTER SESSION SET TIME_ZONE='%s'", timezoneId));
+            return statement.executeQuery(format("SELECT %s FROM dual", expression));
+        }
+
+        @Override
+        public void setup()
+        {
+            try {
+                connection = DriverManager.getConnection(oracleServer.getJdbcUrl(), oracleServer.getUsername(), oracleServer.getPassword());
+                statement = connection.createStatement();
+            }
+            catch (SQLException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public void teardown()
+        {
+            closeQuietly(statement);
+            closeQuietly(connection);
+        }
+
+        @Override
+        public void close()
+        {
+            oracleServer.stop();
+        }
+    }
+
+    private static class PostgresqlReferenceDriver
+            implements ReferenceDriver
+    {
+        private PostgreSQLContainer pgServer;
+        private Connection connection;
+        private Statement statement;
+
+        PostgresqlReferenceDriver()
+        {
+            pgServer = new PostgreSQLContainer("postgres:10.3")
+                    .withDatabaseName("tpch")
+                    .withUsername("test")
+                    .withPassword("test");
```

Comment:
This should be not be necessary.

just replace
```
connection = DriverManager.getConnection(pgServer.getJdbcUrl(), "test", "test");
```

below with
```
connection = pgServer.createConnection("");
```

Reply:
Neat.